pkg/diagnostics/resiliency_monitoring.go

```diff
@@ -80,12 +93,40 @@ func (m *resiliencyMetrics) PolicyLoaded(resiliencyName, namespace string) {
 // PolicyWithStatusExecuted records metric when policy is executed with added status information (e.g., circuit breaker open).
 func (m *resiliencyMetrics) PolicyWithStatusExecuted(resiliencyName, namespace string, policy PolicyType, flowDirection PolicyFlowDirection, target string, status string) {
 	if m.enabled {
+		// Common tags for all metrics
+		commonTags := []interface{}{
+			appIDKey, m.appID, 
+			resiliencyNameKey, resiliencyName, 
+			policyKey, string(policy),
+			namespaceKey, namespace, 
+			flowDirectionKey, string(flowDirection), 
+			targetKey, target, 
+			statusKey, status,
+		}
+
 		_ = stats.RecordWithTags(
 			m.ctx,
-			diagUtils.WithTags(m.executionCount.Name(), appIDKey, m.appID, resiliencyNameKey, resiliencyName, policyKey, string(policy),
-				namespaceKey, namespace, flowDirectionKey, string(flowDirection), targetKey, target, statusKey, status),
+			diagUtils.WithTags(m.executionCount.Name(), commonTags...),
 			m.executionCount.M(1),
 		)
+
+		// Record 4 metrics, one for each cb state, with the active state having a value of 1
+		if policy == CircuitBreakerPolicy {
+			for _, s := range cbStatuses {
+				//Use correct status tag
+				commonTags = append(commonTags[:len(commonTags)-1], s)
+				stateActive := int64(0)
+				if s == status {
+					stateActive = int64(1)
+				}
```

Comment:
we could remove the overwriting of commonTags, and also maybe these two could be a constant (ie `stateActive` and `stateInactive`), and just use:
```
if s == status {
  diagUtils.WithTags(m.circuitbreakerState.Name(), append(commonTags, s),
					m.circuitbreakerState.M(stateActive)
else {
  diagUtils.WithTags(m.circuitbreakerState.Name(), append(commonTags, s),
					m.circuitbreakerState.M(stateInactive)
}
```
this is kind of in the hot path, so a bit of repetition does not hurt to reduce allocations.

check to make sure it is correct